File: tasket/managers/log.py
from tasket.store import LogStore
from tasket.models import Log, LogFilter, LogRequest
from typing import List
from tasket.utils import generate_uid


class LogManager:
    def __init__(self, log_store: LogStore):
        self._log_store = log_store

    def get_by_task(self, task_id: str) -> List[Log]:
        return self._log_store.get_by_task(task_id)

    def get_by_filters(self, task_id: str, filter: LogFilter) -> List[Log]:
        return self._log_store.get_by_filters(task_id, filter)

    def create_log(self, log_req: LogRequest):
        id = generate_uid()
        log = Log(
            id=id,
            task_id=log_req.task_id,
            source=log_req.source,
            content=log_req.content,
            created_at=log_req.created_at,
        )
        self._log_store.insert(log)

File: src/managers/log.go
package managers

import (
	"github.com/J-Obog/tasket/src/types"
)

type LogManager struct {
	logStore     types.ILogStore
	timeProvider types.ITimeProvider
	uuidProvider types.IUUIDProvider
}

func NewLogManager(logStore types.ILogStore, timeProvider types.ITimeProvider, uuidProvider types.IUUIDProvider) *LogManager {
	return &LogManager{
		logStore:     logStore,
		timeProvider: timeProvider,
		uuidProvider: uuidProvider,
	}
}

func (this *LogManager) GetLogsByTask(taskId string) ([]types.Log, error) {
	return this.logStore.GetByTask(taskId)
}

func (this *LogManager) GetLogsByFilter(taskId string, filter types.LogOptions) ([]types.Log, error) {
	return this.logStore.GetByFilter(taskId, filter)
}

func (this *LogManager) CreateLog(taskId string, newLog types.NewLog) error {
	id := this.uuidProvider.UUID()
	now := this.timeProvider.Now()

	log := types.Log{
		Id:        id,
		TaskId:    newLog.TaskId,
		Source:    newLog.Source,
		Content:   newLog.Content,
		CreatedAt: now,
	}

	return this.logStore.Insert(log)
}
